src/core/agent/services/multiSigService.ts

```diff
@@ -716,6 +722,149 @@ class MultiSigService extends AgentService {
     }
     return true;
   }
+
+  async getMultisigParticipants(multisigSignifyName: string) {
+    const members = await this.props.signifyClient
+      .identifiers()
+      .members(multisigSignifyName);
+    const multisigMembers = members["signing"];
+    const allMultisigMemberIdentifiers =
+      await this.identifierStorage.getMultisigMemebersMetadata();
+    const owningMember = multisigMembers.find((member: any) =>
+      allMultisigMemberIdentifiers.some(
+        (identifier) => member.aid === identifier.id
+      )
+    );
+
+    if (!owningMember) {
+      throw new Error(MultiSigService.MEMBER_AID_NOT_FOUND);
+    }
+    const ourIdentifier = allMultisigMemberIdentifiers.find(
+      (identifier) => identifier.id === owningMember.aid
+    );
+    if (!ourIdentifier) {
+      throw new Error(MultiSigService.MEMBER_AID_NOT_FOUND);
+    }
+    return {
+      ourIdentifier,
+      multisigMembers,
+    };
+  }
+
+  async endRoleAuthorization(multisigSignifyName: string) {
+    const { ourIdentifier, multisigMembers } =
+      await this.getMultisigParticipants(multisigSignifyName);
+    const hab = await this.props.signifyClient
+      .identifiers()
+      .get(multisigSignifyName);
+    const aid = hab["prefix"];
+    const results = [];
+    for (const member of multisigMembers) {
+      const eid = Object.keys(member.ends.agent)[0]; //agent of member
+      const stamp = new Date().toISOString().replace("Z", "000+00:00");
+
+      const endRoleRes = await this.props.signifyClient
+        .identifiers()
+        .addEndRole(multisigSignifyName, "agent", eid, stamp);
```

Comment:
Does this mean for our agent, we are allowing all of the eids be authorised? So for a multi-sig of 3, each wallet will authorise all 3 eids?

Reply:
This should only be called once. After that, other participants will join the authorization through the notification.

Reply:
But it's a for loop `for (const member of multisigMembers)` - so each person calls it 3 times?

Reply:
If a member of a multi-sig calls this, it will addEndRole for all members of the multisig. Other members will receive 3 messages and join them.

Reply:
Oh, I forgot about this. However, I don't see them join and send the notifications again and again, let me check again.